Core/include/Acts/EventData/TrackStateProxy.hpp

```diff
@@ -678,7 +678,7 @@ class TrackStateProxy {
 
   /// Set an uncalibrated source link
   /// @param sourceLink The uncalibrated source link to set
-  void setUncalibratedSourceLink(SourceLink sourceLink) requires(!ReadOnly) {
+  void setUncalibratedSourceLink(SourceLink&& sourceLink) requires(!ReadOnly) {
```

Comment:
I can also achieve this with a const reference, which might be a bit easier on clients (controls where the copy happens)

Reply:
I would prefer value or reference over rvalue

Reply:
I'd have thought value + `move` would give ~ same performance as revalue ref, but it's 20-25% slower.

Reply:
that is unexpected - I wonder if the optimizer is tripping somehow

Reply:
Maybe. I'm not sure I'd expect the compiler to completely eliminate the move construction in all cases. With full on rvalue references, we basically are guaranteed to only do a single move construction into the backing vector, and nothing else in the call chain.

Reply:
In principle move also does work on the source where copy only does work on the destination. When looking at the handler the move methods have more lines than the copy methods, so likely for simple types move adds overhead.